LibraBFT/Impl/Handle.agda

```diff
@@ -123,10 +104,52 @@ runHandler st handler = ×-map₂ (outputsToActions {st}) (proj₂ (LBFT-run han
 peerStep : NodeId → NetworkMsg → RoundManager → RoundManager × List (LYT.Action NetworkMsg)
 peerStep nid msg st = runHandler st (handle nid msg 0)
 
-InitAndHandlers : SystemInitAndHandlers ℓ-RoundManager ConcSysParms
-InitAndHandlers = mkSysInitAndHandlers
-                    genesisInfo
-                    initRM
-                    initWrapper
+fakeInitAndHandlers : SystemInitAndHandlers ℓ-RoundManager ConcSysParms
+fakeInitAndHandlers = mkSysInitAndHandlers
+                    fakeGenesisInfo
+                    fakeInitRM
+                    fakeInitWrapper
                     peerStep
 
+------------------------------------------------------------------------------
+-- real initialization
+
+postulate
+  now           : Instant
+  pg            : ProposalGenerator
+
+initEMWithOutput' : Either ErrLog (EpochManager × List Output)
+initEMWithOutput' = do
+  (nf , _ , vss , vv , pe , liws) ← GenKeyFile.create 1 (0 ∷ 1 ∷ 2 ∷ 3 ∷ [])
+  let nfLiwsVssVvPe               = (nf , liws , vss , vv , pe)
+      me                          = 0
+  Init.initialize me nfLiwsVssVvPe now ObmNeedFetch∙new pg
+
+initEMWithOutput : EitherD ErrLog (EpochManager × List Output)
+initEMWithOutput = do
+  (nf , _ , vss , vv , pe , liws) ← fromEither $ GenKeyFile.create 1 (0 ∷ 1 ∷ 2 ∷ 3 ∷ [])
+  let nfLiwsVssVvPe               = (nf , liws , vss , vv , pe)
+      me                          = 0
+  fromEither $ Init.initialize me nfLiwsVssVvPe now ObmNeedFetch∙new pg
+
+initEMWithOutput≡ : ∀ {x} → EitherD-run initEMWithOutput ≡ Right x → initEMWithOutput' ≡ Right x
+initEMWithOutput≡ {x} iewo
+  with GenKeyFile.create 1 (0 ∷ 1 ∷ 2 ∷ 3 ∷ [])
+... | Right (nf , _ , vss , vv , pe , liws)
+  with Init.initialize 0 (nf , liws , vss , vv , pe) now ObmNeedFetch∙new pg
+... | Right y rewrite iewo = cong Right refl
+
+------------------------------------------------------------------------------
+-- TODO : ASK CHRIS : regarding EitherD-run
+
+zzz : EitherD ErrLog ℕ
```

Comment:
Did you check in the right version of this?  This seems to be still in the state right after we figured out that we were confused over the difference between `initEMWithOutput ≡ Right x  and `EitherD-run initEMWithOutput ≡ Right x`.  We have one post-confusion case proved above, but not the 'Left` case, which should be equally trivial.  Also, I think we want (at least) the other direction...

Actually, I just pushed a commit to make clearer what I was about to say.  It shows that the proof that the two functions are equivalent is trivial, as expected/hoped.  All we have to do is the case analysis and rewrite/refl does the rest.

Reply:
Oh yeah - thanks for the update.